src/vm/eventpipe.cpp

```diff
@@ -146,12 +146,15 @@ void EventPipe::Shutdown()
                     Disable(static_cast<EventPipeSessionID>(1ULL << i));
             }
 
-            // Remove EventPipeEventSource first since it tries to use the data structures that we remove below.
-            // We need to do this after disabling sessions since those try to write to EventPipeEventSource.
-            delete s_pEventSource;
-            s_pEventSource = nullptr;
-
-            s_config.Shutdown();
+            // dotnet/corecl: issue 24850: EventPipe shutdown race conditions
+            // Deallocating providers/events here might cause AV if a WriteEvent
+            // was to occur. Thus, we are not doing this cleanup (shame shame shame)
```

Comment:
```suggestion
            // was to occur. Thus, we are not doing this cleanup.
```

This is by design. The whole CoreCLR is on the no cleanup plan.